pkg/cvo/availableupdates.go

```diff
@@ -203,7 +225,7 @@ func calculateAvailableUpdatesStatus(ctx context.Context, clusterID string, tran
 
 	if len(arch) == 0 {
 		return cvoCurrent, nil, nil, configv1.ClusterOperatorStatusCondition{
-			Type: configv1.RetrievedUpdates, Status: configv1.ConditionFalse, Reason: "NoArchitecture",
+			Type: configv1.RetrievedUpdates, Status: configv1.ConditionFalse, Reason: noArchitecture,
 			Message: "The set of architectures has not been configured.",
```

Comment:
Not sure why yet, but this is [tripping e2e-agnostic like][1]:

```
: [sig-instrumentation][Late] Alerts shouldn't report any unexpected alerts in firing or pending state [Suite:openshift/conformance/parallel]
--
Run #0: Failed 5s
{  fail [github.com/onsi/ginkgo@v4.7.0-origin.0+incompatible/internal/leafnodes/runner.go:113]: Jul 28 15:51:34.182: Unexpected alerts fired or pending after the test run:  alert CannotRetrieveUpdates fired for 313 seconds with labels: {endpoint="metrics", instance="10.0.0.8:9099", job="cluster-version-operator", namespace="openshift-cluster-version", pod="cluster-version-operator-7674b7b444-f7rn9", service="cluster-version-operator", severity="warning"} ...
```

and:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/796/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1552651611942162432/artifacts/e2e-agnostic/gather-extra/artifacts/clusterversion.json | jq -r '.items[].status.conditions[] | select(.type == "RetrievedUpdates")'
{
  "lastTransitionTime": "2022-07-28T14:21:43Z",
  "message": "The set of architectures has not been configured.",
  "reason": "NoArchitecture",
  "status": "False",
  "type": "RetrievedUpdates"
}
```

/hold

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/796/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1552651611942162432

Reply:
```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_cluster-version-operator/796/pull-ci-openshift-cluster-version-operator-master-e2e-agnostic/1552651611942162432/artifacts/e2e-agnostic/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-7674b7b444-f7rn9_cluster-version-operator.log | grep 'Payload loaded from'
I0728 14:46:02.143948       1 sync_worker.go:385] Payload loaded from registry.build02.ci.openshift.org/ci-op-7ipcn54g/release@sha256:70dd7baaa8c8550e19dddbe79d75a9d78bc37855974649b8292da0e1dccb4d56 with hash oiJMeRYBbXY=, architecture 
```

not sure if that's relevant, but sounds suspicious.

Reply:
Needed call to `SetArchitecture` in `InitializeFromPayload`